frame/asset-rate/src/lib.rs

```diff
@@ -0,0 +1,247 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Asset Rate Pallet
+//!
+//! - [`Config`]
+//! - [`Call`]
+//!
+//! ## Overview
+//!
+//! The AssetRate pallet provides means of setting conversion rates
+//! for some asset to native balance.
+//!
+//! The supported dispatchable functions are documented in the [`Call`] enum.
+//!
+//! ### Terminology
+//!
+//! * **Asset balance**: The balance type of an arbitrary asset. The network might only know about
+//!   the identifier of the asset and nothing more.
+//! * **Native balance**: The balance type of the network's native currency.
+//! * **Treasury spend**: A payment from the treasury after the corresponding proposal has been
+//!   approved.
+//!
+//! ### Goals
+//!
+//! The asset-rate system in Substrate is designed to make the following possible:
+//!
+//! * Whitelisting assets other than the native currency which can be accepted for Treasury spends.
+//! * Providing a soft conversion for the balance of whitelisted assets to native.
+//! * Updating existing conversion rates.
+//!
+//! ## Interface
+//!
+//! ### Permissioned Functions
+//!
+//! * `create`: Creates a new asset conversion rate.
+//! * `remove`: Removes an existing asset conversion rate.
+//! * `update`: Overwrites an existing assert conversion rate.
+//!
+//! Please refer to the [`Call`] enum and its associated variants for documentation on each
+//! function.
+//!
+//! ### Assumptions
+//!
+//! * Conversion rates will not be used to determine the payment amount in another asset.
+//! * Conversion rates will be used to determine the tier of the spender status, e.g.
+//!   `SmallSpender`, `MediumSpender`, `BigSpender` or `Treasurer`.
```

Comment:
This feels like useful information, but that is too treasury specific, and i would rather have this in the description of this pull request, and not in the pallet docs.

Reply:
I agree and removed all Treasury related documentation from the pallet. This particular comment is already part of this PR's description 🌚